nodejs/eks/cluster.ts

```diff
@@ -216,9 +221,13 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
     } else if (args.instanceRole) {
```

Comment:
There is a third branch of this `if/else if/else if` that also creates an `InstanceProfile`.  I assume we need to update that one as well?

The branching here is a little wild, and probably eventually should get simplified down or refactored a bit - but that can be part of a a follow-up PR.

Reply:
You are right, currently the instanceprofile only seems to get created if a single iam role is passed in among the args. Presumably an instanceprofile (or profiles?) need to get get created in the "if" branch as well? In that branch, a list of roles are passed in. The only use case we currently have is when a single role is passed in, so I didn't pay attention to the other logical branches :)

Reply:
I think we do need to make the change in the other branch as well to be correct here.  We are allowing the user to pass an explicit `InstanceRole` they want to use, but then in some cases will ignore it and create a new one anyway.

Reply:
Ok.. Ill address this shortly. Thanks.

Reply:
just pushed a commit for this.